### PR TITLE
Fix #1129: Backflow code optimised

### LINKED ISSUE #1129: Code optimisation for Backflow Exploration Support
Pr #1036 has managed to finish the exploration back flow support correctly but after finishing that PR, I realised that we can further optimise this code by making the ENUM public and use it at various values to control the flow.

Update:
The code needs to follow approach mentioned here: https://github.com/oppia/oppia-android/pull/2566#issuecomment-772960035

### PR DESCRIPTION
**Pull request by rt4914**
## Explanation

Fixes #1129

This PR optimises the code written in #1036.

The main points of optimisations:
1. Pass enum values instead of numbers from parent activities when opening the exploration.
2. Remove null support from `backflowScreen` variable.

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: app/src/main/java/org/oppia/app/player/exploration/BackflowScreenEnum.kt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package org.oppia.app.player.exploration
+
+/** Enum to decide when an exploration is closed which activity should be displayed. */
+enum class BackflowScreenEnum(val value: Int) {
+  BACKFLOW_SCREEN_LESSONS(0),
+  BACKFLOW_SCREEN_STORY(1),
+  BACKFLOW_SCREEN_DEFAULT(2);
```
Comment (Sarthak2601) [inline app/src/main/java/org/oppia/app/player/exploration/BackflowScreenEnum.kt:7]: I think default should have 0 value here.

File: app/src/main/java/org/oppia/app/player/exploration/ExplorationActivityPresenter.kt [hunk 2]
```diff
@@ -144,11 +150,38 @@ class ExplorationActivityPresenter @Inject constructor(
...
+        )
+      )
+      BackflowScreenEnum.BACKFLOW_SCREEN_DEFAULT.value -> activity.startActivity(
+        TopicActivity.createTopicActivityIntent(
+          context,
+          internalProfileId,
+          topicId
+        )
+      )
+      else -> activity.startActivity(
```
Comment (Sarthak2601) [inline app/src/main/java/org/oppia/app/player/exploration/ExplorationActivityPresenter.kt:178]: Is there a need to have an else statement here when we are already handling everything through the default value ?

**Review by Sarthak2601 — CHANGES_REQUESTED** [review]
Nice implementation. Just a few things here.
Also, when I ran the tests on my system it showed `no tests were found`. It can be a device error as well. Are they passing in yours ? 
Please also take care of the `non-flaky tests`

**Comment by rt4914** [comment]
Earlier this PR was blocked because of test cases issue. Now as this PR needs a lot of changes including suggestions and new changes in current develop, I am closing this PR and this will be created correctly in future.

**Pull request closed without merging**